lib/active_merchant/billing/gateways/litle.rb

```diff
@@ -39,6 +39,88 @@ def purchase(money, payment_method, options = {})
         check?(payment_method) ? commit(:echeckSales, request, money) : commit(:sale, request, money)
       end
 
+      def add_level_2_data(doc, payment_method, options = {})
+        level_2_data = options[:level_2_data]
+        if level_2_data
+          doc.enhancedData do
+            case payment_method.brand
+            when 'visa'
+              doc.salesTax(level_2_data[:sales_tax]) if level_2_data[:sales_tax]
+            when 'master'
+              doc.customerReference(level_2_data[:customer_code]) if level_2_data[:customer_code]
+              doc.salesTax(level_2_data[:total_tax_amount]) if level_2_data[:total_tax_amount]
+              doc.detailTax do
+                doc.taxIncludedInTotal(level_2_data[:tax_included_in_total]) if level_2_data[:tax_included_in_total]
+                doc.taxAmount(level_2_data[:tax_amount]) if level_2_data[:tax_amount]
+                doc.cardAcceptorTaxId(level_2_data[:card_acceptor_tax_id]) if level_2_data[:card_acceptor_tax_id]
+              end
+            end
+          end
+        end
+      end
+
+      def add_level_3_data(doc, payment_method, options = {})
+        level_3_data = options[:level_3_data]
+        if level_3_data
+          doc.enhancedData do
+            case payment_method.brand
+            when 'visa'
+              doc.discountAmount(level_3_data[:discount_amount]) if level_3_data[:discount_amount]
+              doc.shippingAmount(level_3_data[:shipping_amount]) if level_3_data[:shipping_amount]
+              doc.dutyAmount(level_3_data[:duty_amount]) if level_3_data[:duty_amount]
+              doc.detailTax do
+                doc.taxIncludedInTotal(level_3_data[:tax_included_in_total]) if level_3_data[:tax_included_in_total]
+                doc.taxAmount(level_3_data[:tax_amount]) if level_3_data[:tax_amount]
+                doc.taxRate(level_3_data[:tax_rate]) if level_3_data[:tax_rate]
+                doc.taxTypeIdentifier(level_3_data[:tax_type_identifier]) if level_3_data[:tax_type_identifier]
+                doc.cardAcceptorTaxId(level_3_data[:card_acceptor_tax_id]) if level_3_data[:card_acceptor_tax_id]
+              end
+              add_line_item_information_for_level_3_visa(doc, payment_method, level_3_data)
+            when 'master'
+              doc.customerReference :customerReference, level_3_data[:customer_code] if level_3_data[:customer_code]
```

Comment:
❓ 
Why the customerReferencie is repeated here in level_3_data? I thing it is for level II only.

Reply:
<img width="1043" alt="Screen Shot 2022-03-23 at 3 12 28 PM" src="https://user-images.githubusercontent.com/10518225/159787342-208bf6b9-a423-48f0-8195-8a9825d0f56a.png">
As you can see in the attached image, for mastercard the customerReference field is a field used for both level II and level III data

Reply:
Ohh, I can see, ok